rules/terraformrules/terraform_workspace_remote_test.go

```diff
@@ -0,0 +1,200 @@
+package terraformrules
+
+import (
+	"testing"
+
+	hcl "github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint/tflint"
+)
+
+func Test_TerraformWorkspaceRemoteRule(t *testing.T) {
+	cases := []struct {
+		Name     string
+		Content  string
+		Expected tflint.Issues
+	}{
+		{
+			Name: "terraform.workspace in resource with remote backend",
+			Content: `
+terraform {
+	backend "remote" {}
+}
+resource "null_resource" "a" {
+	triggers = {
+		w = terraform.workspace
+	}
+}`,
+			Expected: tflint.Issues{
+				{
+					Rule:    NewTerraformDeprecatedInterpolationRule(),
```

Comment:
`NewTerraformWorkspaceRemoteRule`...

It seems that tests are passed because `AssertIssues` ignores the `Rule` field.

Reply:
Whoops, obviously did some copy-pasting here. Will send a fix for `AssertIssues` as well.